### DISCUSSION TITLE
<select multiple> in the multiselect filter

### DISCUSSION THREAD

**Discussion by alukardua1** [rappasoft/laravel-livewire-tables#1007] [Q&A]
how can I set <select multiple> instead of checkboxes in the multiselect filter?

**Comment by lrljoe** [comment]
Not part of the core package currently, you'd need to publish the views, and add "multiple" to the select element in components/tools/filters/select.blade.php or replace multi-select with something similar.

You can see an example of how to approach it in my pull request here, not sure if/when the pull request will be accepted!
[https://github.com/rappasoft/laravel-livewire-tables/pull/1011](https://github.com/rappasoft/laravel-livewire-tables/pull/1011)

**Reply by BenOussama180** [reply]
is there anyway to make it look pretty ? such as slimselect or tomselect with a searchable dropdown ?

**Reply by lrljoe** [reply]
Yes.  I've got a select2 filter that does what you're after.  Let me see what @rappasoft thinks about publishing plugins/add-ons for this.

**Reply by BenOussama180** [reply]
could u share you multi delect dropdown blade file with me please

**Reply by lrljoe** [reply]
For the select2 variant it's more than just the blade I'm afraid as there is some backend stuff for pulling out the data. I'll look at getting it and the other filters packaged up once @rappasoft gives a thumbs up.

Won't be more than a day or two if/once he gives that.

For the straight forward multi select drop-down, this is part of core and docs are available on the website.

**Reply by BenOussama180** [reply]
hmm i undrestand, but yea i ave a select drop-down of coountries, and you know we have a lot , the user cant look them up manully there should be a search for it

**Reply by rappasoft** [reply]
> Yes. I've got a select2 filter that does what you're after. Let me see what @rappasoft thinks about publishing plugins/add-ons for this.

I have no problem with third-party add-ons as long as they're not dependencies. So the docs will have to explain that those need to be installed as well.

**Reply by lrljoe** [reply]
> hmm i undrestand, but yea i ave a select drop-down of coountries, and you know we have a lot , the user cant look them up manully there should be a search for it 
> 

Just so I know which version of the filter to focus on first, do you want searching of the db, or searching of an existing collection/array of values. 
I'd normally recommend searching the collection/array client side where it's not a massive number of records as it prevents the db from being overloaded.  But let me know which you prefer and I'll try to get something up in the next couple of days.  Need to write some docs and clean up my ranty code a bit.

**Reply by BenOussama180** [reply]
@lrljoe , one question isnt select2 uses jquery underthehood ? cant you make somthing that is made with alpine js or vanilla js ?

**Reply by lrljoe** [reply]
> @lrljoe , one question isnt select2 uses jquery underthehood ? cant you make somthing that is made with alpine js or vanilla js ?

Yep, I was using select2 as a generic term apologies. The filter uses alpinejs, no additional requirements/dependencies.

I just need to clean up the code, and write some docs and test the bootstrap version properly as I've done most of the testing in a tailwind version

Question is still which version of the filter you'd prefer as I'll focus on writing up docs for that one first!

**Reply by lrljoe** [reply]
As I'm still slacking on finishing the documentation, here's a quick preview with most of the filters working across Tailwind 2/3, Bootstrap 4/5.
[http://tabledemo.lowerrocklabs.com/](http://tabledemo.lowerrocklabs.com/)
I've yet to finish tidying things up/making things look nicer, but gives you an idea of functionality.

**Comment by alukardua1** [comment]
I figured it out, I just can't figure out how to properly pass an array of options and make a call to builder

**Reply by lrljoe** [reply]
So array of options are passed via:
`->options([])`

Then your builder is accessed via:
```
->filter(function (Builder $builder, array $values) {
     $builder->YOURCRITERIAHERE
  });
```
  
Here's a very basic example where you're only including those records that have a "Team" that you've selected from the list, you'd probably want to build your array from your database though, and you can do this in the configure() or mount() functions.

```
/**
 * @return array<mixed>
 */
public function filters(): array
{
  $this->filterTeamsArray = ['1' => 'Team 1', '2' => 'Team 2'];
   return [
               MultiSelectFilter::make('Teams')
               ->options($this->filterTeamsArray)
               ->filter(function (Builder $builder, array $values) {
                         $builder->whereHas('teams', fn ($query) => $query->whereIn('teams.id', $values));
               }),
   ];
}
```

**Comment by lrljoe** [comment]
Now part of core.
